master/buildbot/test/unit/util/test_service.py

```diff
@@ -163,13 +163,13 @@ def test_compare(self):
         b2 = self.makeService(attach_to_master=False, name='b', serviceid=21)  # same args as 'b1'
         b3 = self.makeService(attach_to_master=False, name='b', serviceid=20)  # same id as 'a'
 
-        self.assertTrue(a == a)  # pylint: disable=comparison-with-itself
+        self.assertTrue(a == a)  # noqa: PLR0124
```

Comment:
Is it possible to configure ruff to allow full name of the warning in places like here? The warning ID is completely nondescriptive and thus is blocker for retiring pylint.

Reply:
Turns out like this is not implemented in ruff. But it will be there eventually.

Reply:
I think it makes sense to have this PR in before ruff supports human-friendly rule names (https://github.com/astral-sh/ruff/issues/1773). The speed benefits outweigh small additional confusion which will be addressed in the future anyway.

Reply:
I find those short labels easy to lookup in the ruff documentation: https://docs.astral.sh/ruff/rules/
Them being shorter actually makes them easier to copy.

Reply:
Agreed, but when reading code it's completely unclear what these numeric codes mean. Usually code is read far more times than it is written.

In any case, this PR can go after merge conflicts are resolved (do we need to rebase and just drop some commits?). We can revisit the rule names later.

Reply:
Finally fixed those merge conflicts.